sympy/core/power.py

```diff
@@ -263,6 +263,11 @@ def __new__(cls, b, e, evaluate=None):
                     b = -b
                 elif e.is_odd:
                     return -Pow(-b, e)
+            if b.is_idempotent:
+                if e is S.Zero:
```

Comment:
The exponent needs to be positive. If `x*x = x`, and `x*y = 1`, then `x*x*y = x*1 = x*y = x = 1`, so idempotents are only invertible if they are the identity.

For example, this matrix example from [Wikipedia](https://en.wikipedia.org/wiki/Idempotent_matrix), `Matrix([[2, -2, -4], [-1, 3, 4], [1, -2, -3]])` is not invertible.

Reply:
Corrected